modules/data_release/ajax/AddPermission.php

```diff
@@ -12,25 +12,53 @@
  *  @link     https://github.com/aces/Loris
  */
 
-$DB =& Database::singleton();
-
+$factory  = \NDB_Factory::singleton();
+$settings = $factory->settings();
+$DB       = $factory->database();
 if ($_POST['action'] == 'addpermission') {
     $userid          = $_POST['userid'];
     $data_release_id = $_POST['data_release_id'];
-    $success         = $DB->insert(
-        'data_release_permissions',
-        array(
-         'userid'          => $userid,
-         'data_release_id' => $data_release_id,
-        )
-    );
 
-    $factory  = NDB_Factory::singleton();
-    $settings = $factory->settings();
+        $result = $DB->pselectOne(
+            "SELECT COUNT(*) FROM data_release_permissions
+             WHERE userid = :userid and
+             data_release_id = :data_release_id",
+            array(
+             'userid'          => $userid,
+             'data_release_id' => $data_release_id,
+            )
+        );
+
+        $baseURL = $settings->getBaseURL();
+        if ($result != '1') {
+            $success = $DB->insert(
+                'data_release_permissions',
+                array(
+                 'userid'          => $userid,
+                 'data_release_id' => $data_release_id,
+                )
+            );
+
+            header("Location: {$baseURL}/data_release/?addpermissionSuccess=true");
```

Comment:
If this is an ajax script why is it redirecting instead of returning the value that javascript expects to parse?